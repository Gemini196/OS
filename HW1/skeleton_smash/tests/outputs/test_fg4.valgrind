==45358== Memcheck, a memory error detector
==45358== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==45358== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==45358== Command: /home/student/Desktop/OS/HW1/skeleton_smash/smash
==45358== Parent PID: 45354
==45358== 
==45358== 
==45358== FILE DESCRIPTORS: 6 open at exit.
==45358== Open file descriptor 1: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg4.out
==45358==    at 0x54EF957: dup2 (syscall-template.S:78)
==45358==    by 0x112CC9: main (smash.cpp:41)
==45358== 
==45358== Open file descriptor 0:
==45358==    at 0x54EF957: dup2 (syscall-template.S:78)
==45358==    by 0x112C77: main (smash.cpp:33)
==45358== 
==45358== Open file descriptor 5:
==45358==    at 0x54EF927: dup (syscall-template.S:78)
==45358==    by 0x112E17: __static_initialization_and_destruction_0(int, int) (smash.cpp:9)
==45358==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==45358==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==45358==    by 0x5400C17: (below main) (libc-start.c:266)
==45358== 
==45358== Open file descriptor 4: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg4.out
==45358==    at 0x54EF927: dup (syscall-template.S:78)
==45358==    by 0x112E07: __static_initialization_and_destruction_0(int, int) (smash.cpp:8)
==45358==    by 0x112E33: _GLOBAL__sub_I_stdout_dp (smash.cpp:57)
==45358==    by 0x112EAC: __libc_csu_init (in /home/student/Desktop/OS/HW1/skeleton_smash/smash)
==45358==    by 0x5400C17: (below main) (libc-start.c:266)
==45358== 
==45358== Open file descriptor 3: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg4.valgrind
==45358==    <inherited from parent>
==45358== 
==45358== Open file descriptor 2: /home/student/Desktop/OS/HW1/skeleton_smash/tests/outputs/test_fg4.err
==45358==    <inherited from parent>
==45358== 
==45358== 
==45358== HEAP SUMMARY:
==45358==     in use at exit: 0 bytes in 0 blocks
==45358==   total heap usage: 66 allocs, 66 frees, 85,293 bytes allocated
==45358== 
==45358== All heap blocks were freed -- no leaks are possible
==45358== 
==45358== For counts of detected and suppressed errors, rerun with: -v
==45358== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
