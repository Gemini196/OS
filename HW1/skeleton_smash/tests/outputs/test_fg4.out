smash> submitted by Eran
smash> smash> smash> smash> smash> smash> sleep 200& : 46986
smash: got ctrl-Z
smash: process 46986 was stopped
smash> [1] sleep 100& : 46985 2 secs
[2] sleep 200& : 46986 0 secs (stopped)
smash> sleep 100& : 46985
smash: got ctrl-C
smash: process 46985 was killed
smash> [2] sleep 200& : 46986 1 secs (stopped)
smash> smash> sleep 300& : 47153
smash: got ctrl-C
smash: process 47153 was killed
smash: got ctrl-C
smash> smash: got ctrl-Z
smash: process 0 was stopped
smash: got ctrl-Z
[2] sleep 200& : 46986 3 secs (stopped)
[3] sleep 300& : 0 0 secs (stopped)
smash> smash: sending SIGKILL signal to 2 jobs:
46986 : sleep 200&
0 : sleep 300&
